cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_data_resources.go

```diff
@@ -203,9 +203,31 @@ func (s *Server) getApp(ctx context.Context, cluster, namespace, identifier stri
 
 //  List of resources getters
 
-// getPkgs returns the list of packages for the given cluster and namespace
-func (s *Server) getPkgs(ctx context.Context, cluster, namespace string) ([]*datapackagingv1alpha1.Package, error) {
-	return s.getPkgsWithFieldSelector(ctx, cluster, namespace, "")
+// getPkgs requests the packages for the given cluster and namespace and sends
+// them to the channel to be processed immediately, closing the channel
+// when finished or when an error is returned.
+func (s *Server) getPkgs(ctx context.Context, cluster, namespace string, ch chan<- *datapackagingv1alpha1.Package) error {
```

Comment:
This function wasn't actually used at all, so I updated it to do what I want. Importantly, it does not iterate all the results into an in-memory slice, but rather sends each package down the return channel as it iterates (so we can process them while it continues iterating).

The other important point here is, as mentioned in the comment, that this function guarantees to close the channel, so we can be sure when it has completed (or returned with an error).